### PR TITLE
test: Airdrop to contract tests

### PR DESCRIPTION
**Pull request by ibankov**
**Description**:
This PR covers AIRDROP_TO_CONTRACT_04,05,06,07

**Related issue(s)**:
[#15123](https://github.com/hashgraph/hedera-services/issues/15123)
[#15125](https://github.com/hashgraph/hedera-services/issues/15125)
[#15127](https://github.com/hashgraph/hedera-services/issues/15127)
[#15128](https://github.com/hashgraph/hedera-services/issues/15128)
Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM Thanks @ibankov ! Just a minor comment, can be changed in later PRs

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropTest.java [hunk 1]
```diff
@@ -1966,5 +1970,126 @@ final Stream<DynamicTest> multipleTokensOnlyNFTIsAssociated() {
...
+                    getTxnRecord("airdropToContractTxn")
+                            .hasPriority(recordWith()
+                                    .pendingAirdrops(includingNftPendingAirdrop(
+                                            movingUnique(nftWithCustomFee, 1L).between(OWNER, collectorContract)))),
+                    getAccountBalance(collectorContract).hasTokenBalance(nftWithCustomFee, 0)));
+        }
+
+        @HapiTest
+        @DisplayName("with custom fee to different collector when exempt")
+        final Stream<DynamicTest> customFeeToDifferentCollectorWhenExempt() {
```
[the remark this answers is not in the document]
Comment (ibankov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropTest.java:2020]: Thanks, will address this in separate PR.

**Review by iwsimon — APPROVED** [review]
LGTM, thanks @ibankov

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @ibankov

**Pull request merged**